pkg/gardenadm/botanist/network_test.go

```diff
@@ -0,0 +1,103 @@
+// SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package botanist_test
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	fakeclient "sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	resourcesv1alpha1 "github.com/gardener/gardener/pkg/apis/resources/v1alpha1"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	fakekubernetes "github.com/gardener/gardener/pkg/client/kubernetes/fake"
+	. "github.com/gardener/gardener/pkg/gardenadm/botanist"
+	"github.com/gardener/gardener/pkg/gardenlet/operation"
+	botanistpkg "github.com/gardener/gardener/pkg/gardenlet/operation/botanist"
+	"github.com/gardener/gardener/pkg/gardenlet/operation/shoot"
+)
+
+var _ = Describe("Network", func() {
+	var (
+		ctx       context.Context
+		namespace = "kube-system"
+
+		b *AutonomousBotanist
+	)
+
+	BeforeEach(func() {
+		ctx = context.Background()
+
+		b = &AutonomousBotanist{
+			Botanist: &botanistpkg.Botanist{
+				Operation: &operation.Operation{
+					Shoot: &shoot.Shoot{
+						ControlPlaneNamespace: namespace,
+					},
+					SeedClientSet: fakekubernetes.
+						NewClientSetBuilder().
+						WithClient(fakeclient.NewClientBuilder().WithScheme(kubernetes.SeedScheme).Build()).
+						Build(),
+				},
+			},
+		}
+	})
+
+	Describe("#IsPodNetworkAvailable", func() {
+		var (
+			managedResource *resourcesv1alpha1.ManagedResource
+		)
+
+		BeforeEach(func() {
+			managedResource = &resourcesv1alpha1.ManagedResource{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "shoot-core-coredns",
+					Namespace: namespace,
+				},
+			}
+		})
+
+		It("should return false because the ManagedResource does not exist", func() {
```

Comment:
A relevant scenario, which is handled in the code, but not in any test is the case that the `ManagedResource` crd has not yet been deployed. Should we add a test for it as well? WDYT?

Reply:
Do you have a proposal for writing such a test with the fake client? I'm not sure whether it is reasonable to go down the mock client-path just for this corner case. However, if there is an idea how to realize this with the fake client, I'm happy to add it.